plugins/json/README.md

```diff
@@ -0,0 +1,197 @@
+
+The easiest way to test the plugin is refering to the example section, that provides a full somewhat realistic C simulation code, where PDI is enabled and the JSON plugin is used. This one can be found at \ref PDI_example
+
+To use it, first compile using CMake with `BUILD_TESTING` set to `ON`, then execute : 
+```bash
+build/PDI_EXAMPLE/PDI_example_C example/json.yml
+```
+
+<!-- ### Example output -->
+
+## Full configuration file example {#json_config_example}
+
+\snippet "../../example/json.yml" example_config_json_plugin
+
+## On going work / Limitations {#json_on_going_work}
+
+- [ ] Adding support for reading variables. 
+    Currently, the JSON plugin doesn't support reading variables, only writing them. To cover more use cases, the JSON plugin should be able to read variables as well as write them.
+- [x] Improve strings handling. Concatenate arrays of chars to have "var": "value" instead of "var": ["v", "a", ...].
+- [x] Added full support for types that can be recursive, e.g. arrays of arrays, arrays within structures etc. 
+- [ ] The ability to create an on-the-go structures for several variables to be produced in a single JSON object
+- [x] Continuous integration: More test units coverage
+
+## The JSON plugin for custom formating {#json_custom_format}
+
+Initially, the aim of this plugin was to enable the writing of customized file formats. To achieve this, the template systems are logical and first-hand, in particular `mustache`. The latter, however, allows you to take JSON as input and format it.
+
+For the sake of the example, let's imagine the following code, creating which performs some kind of heat transfer simulation.
+
+### The PDI exposes
+
+<!-- \image html json_simulation_code_light.png width=800px -->
+
+```C
+    PDI_multi_expose("init",
+        "simulation_name", &cfg.simulation_name, PDI_OUT,
+        "max_steps", &cfg.max_steps, PDI_OUT,
+        "mesh_config", &cfg.mesh_config, PDI_OUT,
+        "rank", &rank, PDI_OUT,
+        NULL) ;
+
+    // main loop
+    for (int step = 0; step < nb_steps; ++step) {
+        do_compute(temp, MPI_COMM_WORLD) ;
+
+        // share data at every iteration
+        PDI_multi_expose("iter",
+            "step", &step, PDI_OUT,
+            "temp", temp, PDI_OUT,
+            NULL) ;
+        MPI_Barrier(MPI_COMM_WORLD);
+        ...
+    }
+```
+
+### The related YAML configuration
+<!-- \image html simulation_yaml_light.png width=800px -->
+
+```yaml
+types: # [...] including config_t description
+    metadata: {rank: int, step: int}
+    data:
+    simulation_name: { type: array, subtype: char, size: 512 }
+    max_steps: int
+    mesh_config:
+    type: struct
+        members:
+        - dimensions: { type: array, subtype: int, size: 3}
+        - spacings: { type: array, subtype: int, size: 3}
+
+        temp: # the main temperature field
+        - type: array
+        - subtype: double
+        - size: 'Smesh_config.dimensions'
```

Comment:
I suggest to reindent this types to have a better understanding of what is inside in the data.
I correct also 'Smesh_config.dimensions' by '$mesh_config.dimensions'
```suggestion
```yaml
types: # [...] including config_t description
    metadata: {rank: int, step: int}
    data:
        simulation_name: { type: array, subtype: char, size: 512 }
        max_steps: int
        mesh_config:
           type: struct
           members:
           - dimensions: { type: array, subtype: int, size: 3}
           - spacings: { type: array, subtype: int, size: 3}

        temp: # the main temperature field
        - type: array
        - subtype: double
        - size: '$mesh_config.dimensions'
```

Reply:
thanks